conans/client/conf/__init__.py

```diff
@@ -66,6 +66,7 @@
         baremetal:
         VxWorks:
             version: ["7"]
+        Dos:
```

Comment:
should it be MS-DOS instead of generic DOS which is just Disk Operation System (AFAIK there were many implementations, such as TR-DOS, DR-DOS, MSX-DOS, etc, and I am not sure if they were binary compatible)

Reply:
There is also Freedos, which is still maintained.
I'm proposing a simple Dos here as a common base.
I don't know exactly, but I suppose there is some compatibility.
I think some/a lot syscalls are shared

A subsystem-like option might be needed, as you can build dos applications in multiple configurations:
e.g. LE executable, LX executable, dos4g extender, dos32a extender, ...
I'm going to ask watcom upstream what they think of this.

Reply:
from what I remember from these times, there were some interrupts provided by MS-DOS (e.g. int 21h), but these were way too limited, and almost every program practically used BIOS interrupts (such as int 10h).
for binaries, yes, the most common formats were MZ and COM. LE/LX formats were designed for OS/2, and NE was designed for Win 3.x, but they were used by various DOS extenders (probably, they all were incompatible).
it has to be checked in watcom compiler docs what exactly can it generate.

Reply:
One thing I'm also unsure about is the architecture.
Watcom can build 16- and 32-bit executables. So should such an option be added? And if it is needed, how should we name it?
Here's a quick overview of what archs are supported + compiler flags to use: https://wiki.archlinux.org/title/Open_Watcom

Reply:
yes, we probably will have to add x86_16 (or whatever name can be used to distinguish 16-bit from 32-bit x86, quick googling doesn't find any widespread name)
dos4gw (and other extenders) probably should be sub-systems.

so it may look like:
```
os: MSDOS
arch: x86, x86_16
subsystem: [None, "dos4gw"]
```